openmmml/models/nequippotential.py

```diff
@@ -0,0 +1,198 @@
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS, CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+from openmmml.mlpotential import MLPotential, MLPotentialImpl, MLPotentialImplFactory
+import openmm
+from typing import Iterable, Optional, Union, Tuple
+from openmmml.models.utils import simple_nl
+
+class NequIPPotentialImplFactory(MLPotentialImplFactory):
+    """This is the factory that creates NequipPotentialImpl objects."""
+
+    def createImpl(self, name: str, model_path: str, distance_to_nm: float, energy_to_kJ_per_mol: float, atom_types: Optional[Iterable[int]]=None, **args) -> MLPotentialImpl:
+        return NequIPPotentialImpl(name, model_path, distance_to_nm, energy_to_kJ_per_mol, atom_types)
+
+class NequIPPotentialImpl(MLPotentialImpl):
+    """This is the MLPotentialImpl implementing the NequIP potential.
+
+    The potential is implemented using NequIP to build a PyTorch model.
+    A TorchForce is used to add it to the OpenMM System. Note that you must
+    provide a deployed model. No general purpose model is available.
+
+    There are three required keyword arguments
+
+    model_path: str
+        path to deployed NequIP model
+    distance_to_nm: float
+        conversion constant between the nequip model distance units and OpenMM units (nm)
+    energy_to_kJ_per_mol: float
+        conversion constant between the nequip model energy units and OpenMM units (kJ/mol)
+
+    for example
+
+    >>> potential = MLPotential('nequip', model_path='example_model_deployed.pth',
+                        distance_to_nm=0.1, energy_to_kJ_per_mol=4.184)    
+    
+    There is one optional keyword argument that lets you specify the nequip atom type of 
+    each atom. Note that by default this potential uses the atomic number to map the NequIP atom type. 
+    This will work if you trained your NequIP model using the standard `chemical_symbols` option. If you
+    require more flexibility you can use the atom_types argument. It must be a list containing an 
+    integer specifying the nequip atom type of each particle in the system.
+
+    atom_types: List[int]
+
+
+    """
+
+    def __init__(self, name, model_path, distance_to_nm, energy_to_kJ_per_mol, atom_types):
```

Comment:
Does NequIP have default units it uses, or are the units arbitrary?  If it has a preferred set of units, we can put the conversion factors here as default values.

Reply:
Based on the discussion at https://github.com/mir-group/nequip/issues/288#issuecomment-1431574566, it appears that NequIP/Allegro is entirely agnostic to units and preserves those of the training dataset. I think it's better for users to receive a `TypeError` indicating missing arguments rather than potentially proceeding with incorrect conversions.